src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TaskStatusCenter;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Diagnostics
+{
+    [Export(typeof(DiagnosticProgressReporter))]
+    internal class DiagnosticProgressReporter
+    {
+        private readonly IVsTaskStatusCenterService _taskCenterService;
+        private readonly IDiagnosticService _diagnosticService;
+        private readonly TaskHandlerOptions _options;
+
+        // this fields are never accessed concurrently
+        private TaskCompletionSource<bool> _currentTask;
+        private string _lastMessage;
+
+        // this is only field that is shared between 2 events streams (IDiagnosticService and ISolutionCrawlerProgressReporter)
+        // and can be called concurrently.
+        // volatile to make sure value is shared properly between threads
```

Comment:
:bulb: The first two lines of this comment are very helpful, but the third is just repeating what `volatile` always means (i.e. the third line can be removed to simplify the comment without any real downside that I see, IMO of course)

Reply:
sure